### ISSUE TITLE
Investigate possibility of SQL constraints

### ISSUE THREAD

**Issue by xZise** [espinosaluis/TrackMeViewer#80]
I haven't done those in some time but afaik it should be possible to say that each row in `positions` needs to have either a `FK_Trips_ID` which is not null or a `FK_Users_ID` which is not null (both at the same time doesn't make sense). And if they aren't null they must be a valid ID in the corresponding table. Also `FK_Icons_ID` needs to be a valid icon ID from the `icons` table (or null).

Additionally deleting a trips row should then delete the positions linked to it, as well as deleting a user should delete their trips and positions.

**Comment by espinosaluis** [comment]
Yes, I have been postponing adding new constraints for a long time and what you say it makes sense. I need to revisit the whole database and add rules to keep all the data consistent.
